### PR TITLE
[BAD-140] - Compile HDP 2.2 shim

### PR DESCRIPTION
**Pull request by andrei-viaryshka**

### CODE REVIEW

**Comment by mbradbury2014** [comment]
@brosander , @mattyb149 Could one of you guys please take a look?

File: build.xml [hunk 1]
```diff
@@ -14,7 +14,7 @@
   <import file="build-res/subfloor-pkg.xml"/>
 
   <!-- Every non-api shim module. These will be compiled/resolved after api. -->
-  <property name="shim-non-api-modules" value="hive-jdbc,hadoop-20,cdh47,cdh52,hdp21,mapr31,mapr401" />
+  <property name="shim-non-api-modules" value="hive-jdbc,hadoop-20,cdh47,cdh52,hdp21,hdp22,mapr31,mapr401" />
```
Comment (mattyb149) [inline build.xml]: should hdp22 replace hdp21 in the list?

File: hdp22/src/org/pentaho/hadoop/shim/hdp22/authentication/HadoopNoAuthConsumer.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  public HadoopNoAuthConsumer( Properties props ) {
+    // Noop
+  }
+
+  @Override
+  public HadoopAuthorizationService consume( NoAuthenticationAuthenticationProvider authenticationProvider )
+    throws AuthenticationConsumptionException {
+    return new NoOpHadoopAuthorizationService();
+  }
+}
```
Comment (mattyb149) [inline hdp22/src/org/pentaho/hadoop/shim/hdp22/authentication/HadoopNoAuthConsumer.java]: We REALLY should move this guy up into common so we don't have a copy for every shim

File: hdp22/src/org/pentaho/hadoop/shim/hdp22/authorization/HasHadoopAuthorizationService.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+******************************************************************************/
+
+package org.pentaho.hadoop.shim.hdp22.authorization;
+
+public interface HasHadoopAuthorizationService {
+  public void setHadoopAuthorizationService( HadoopAuthorizationService hadoopAuthorizationService ) throws Exception;
+}
```
Comment (mattyb149) [inline hdp22/src/org/pentaho/hadoop/shim/hdp22/authorization/HasHadoopAuthorizationService.java]: This also should be moved to common

File: hdp22/src/org/pentaho/hadoop/shim/hdp22/delegating/DelegatingHBaseConnection.java [hunk 4]
```diff
@@ -0,0 +1,532 @@
...
+  @Override
+  public void addColumnFilterToScan( ColumnFilter cf, HBaseValueMeta columnMeta, VariableSpace vars, boolean matchAny )
+    throws Exception {
+    delegate.addColumnFilterToScan( cf, columnMeta, vars, matchAny );
+  }
+
+  /*
+   * (non-Javadoc)
+   *
+   * @see org.pentaho.hadoop.shim.mapr31.delegatingShims.HBaseConnectionInterface#addColumnToScan(java.lang.String,
```
Comment (mattyb149) [inline hdp22/src/org/pentaho/hadoop/shim/hdp22/delegating/DelegatingHBaseConnection.java]: Should not refer to mapr31 shim in Javadoc (many instances of this throughout the file)

**Comment by mattyb149** [comment]
It would be helpful for reviewing these kinds of PRs if we knew what (if anything) changed from the last version of the shim (hdp21 in this case). If it was mostly a cut-n-paste exercise and updated versions in ivy.xml then fine, but if additional code was needed (due to API changes, e.g.) it would be good to call those out so we can review them in more detail.  Thanks!

File: build.xml (hunk 1 above)
Comment (andrei-viaryshka) [inline build.xml]: I'm not sure about this. AFAIK, we just adding hdp22 shim for now.
@mbradbury2014 Should we just add the hdp22 shim or replace hdp21?

**Comment by andrei-viaryshka** [comment]
It was mostly copy-n-paste with updating versions.
The only difference from original hdp21 is ported fix from cdh52 related to GruntParser API change.

**Comment by mattyb149** [comment]
Thanks much!  We don't have to do the refactor of common code at this point, as we are focusing on the rearchitecture of the shim stuff anyway

**Pull request merged**